runtime/Cpp/runtime/src/UnbufferedCharStream.cpp

```diff
@@ -101,7 +101,7 @@ size_t UnbufferedCharStream::LA(ssize_t i) {
     return EOF;
   }
 
-  if (_data[(size_t)index] == (uint32_t)EOF) {
+  if (_data[(size_t)index] == 0xFFFF) {
```

Comment:
Oh damn, you are right. I somehow thought that was ATN related (and hence a 16bit value). But that's from the unbuffered char stream. Need to review that.